index.js

```diff
@@ -8,64 +9,44 @@ function generateWorkerName(browser) {
     return [browser.os, browser.os_version, browser.browser, browser.browser_version, browser.device].join(' ').trim();
 }
 
-exports.createWorkers = function createWorkers(browsers, callback, workerIds) {
-    if (browsers.length) {
-        var browser = browsers.shift(),
-            workerName = browser.worker_name || generateWorkerName(browser);
+exports.createWorkers = function createWorkers(browsers, callback) {
+    async.map(browsers, function(browser, callback) {
+        var workerName = browser.worker_name || generateWorkerName(browser),
+            browserInfo = {
+                os: browser.os,
+                os_version: browser.os_version,
+                browser: browser.browser,
+                browser_version: browser.browser_version,
+                device: browser.device,
+                url: 'http://' + config.TestCafe.hostname + ':' + config.TestCafe.controlPanelPort +
+                     '/worker/add/' + encodeURI(workerName)
+            };
 
-        workerIds = workerIds || [];
-
-        delete browser.worker_name;
-
-        browser.url = 'http://' + config.TestCafe.hostname + ':' + config.TestCafe.controlPanelPort + 
-            '/worker/add/' + encodeURI(workerName);
-
-        bsClient.createWorker(browser, function(err, worker) {
-            if (!err) {
-                workerIds.push(worker.id);
-                createWorkers(browsers, callback, workerIds);
-            } else
-                callback && callback(workerIds, err);
+        bsClient.createWorker(browserInfo, function(err, worker) {
+            callback(err, !err && worker.id);
         });
-    } else
-        callback && callback(workerIds);
+    }, function(err, workerIds) {
+        callback && callback(workerIds, err);
```

Comment:
It's a strange for me that we returns `err` as the second `callback` argument and `workerIds` as the first. Usually, it's `callback(err, res)`. But I see, it was before this commit. If we will change this it's a breaking change. /cc @inikulin 

lgtm for this commit changes